### PR TITLE
libselinux: Add CPPFLAGS to Makefile

### PR DESCRIPTION
**Pull request by samsam2310**
Add CPPFLAGS to Makefile to allow users change the flags of
preprocessor.
We offen use CFLAGS for compiler flags and use CPPFLAGS for
preprocessor.

Signed-off-by: Chung-Sheng Wu <chungsheng@google.com>

### CODE REVIEW

File: libselinux/src/Makefile [hunk 1]
```diff
@@ -149,17 +149,17 @@ pywrap: all selinuxswig_python_exception.i
...
-	$(CC) $(CFLAGS) $(LDFLAGS) -L. -shared -o $@ $^ -lselinux $(RUBYLIBS)
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -L. -shared -o $@ $^ -lselinux $(RUBYLIBS)
 
 $(LIBA): $(OBJS)
 	$(AR) rcs $@ $^
 	$(RANLIB) $@
 
 $(LIBSO): $(LOBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ $(PCRE_LDLIBS) $(FTS_LDLIBS) -ldl -Wl,$(LD_SONAME_FLAGS)
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ $(PCRE_LDLIBS) $(FTS_LDLIBS) -ldl -Wl,$(LD_SONAME_FLAGS)
```
Comment (vapier) [inline libselinux/src/Makefile:162]: you don't need CPPFLAGS when linking (only)

**Comment by samsam2310** [comment]
Sent a mail for the patch. Will close this once it get reviewed.

File: libselinux/src/Makefile [hunk 2]
```diff
@@ -149,10 +149,10 @@ pywrap: all selinuxswig_python_exception.i
 rubywrap: all $(SWIGRUBYSO)
 
 $(SWIGRUBYLOBJ): $(SWIGRUBYCOUT)
-	$(CC) $(CFLAGS) $(SWIG_CFLAGS) $(RUBYINC) -fPIC -DSHARED -c -o $@ $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(SWIG_CFLAGS) $(RUBYINC) -fPIC -DSHARED -c -o $@ $<
 
 $(SWIGRUBYSO): $(SWIGRUBYLOBJ)
-	$(CC) $(CFLAGS) $(LDFLAGS) -L. -shared -o $@ $^ -lselinux $(RUBYLIBS)
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -L. -shared -o $@ $^ -lselinux $(RUBYLIBS)
```
Comment (vapier) [inline libselinux/src/Makefile:155]: don't use `$(CPPFLAGS)` when linking

**Comment by samsam2310** [comment]
Updated. But created a new PR #398  to merge to main branch.

**Pull request closed without merging**